cmd/attest.go

```diff
@@ -0,0 +1,218 @@
+package cmd
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"strconv"
+
+	"github.com/spf13/cobra"
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm-tools/proto/attest"
+	"google.golang.org/protobuf/encoding/prototext"
+	"github.com/google/go-tpm/tpm2"
+	"google.golang.org/protobuf/proto"	
+)
+var (
+	key      string
+	nonce    []byte
+	teenonce []byte
+	outputText string
+	gceZone  string
+    gceProjectID     string
+    gceProjectNumber uint64
+    gceInstanceName  string
+    gceInstanceID    uint64
+
+)
+
+var attestationKeys = map[string]map[tpm2.Algorithm]func (rw io.ReadWriter)(*client.Key,error)  {
+	"AK":{
+		tpm2.AlgRSA: client.AttestationKeyRSA,
+		tpm2.AlgECC: client.AttestationKeyECC,
+	},
+	"gceAK":{
+		tpm2.AlgRSA:client.GceAttestationKeyRSA,
+		tpm2.AlgECC:client.GceAttestationKeyECC,
+	},
+}
+
+var attestCmd = &cobra.Command{
+	Use:   "attest",
+	Short: "Create a remote attestation report",
+	Long: `Gather information for remote attestation.
+
+The Attestation report contains a quote on all available PCR banks ,a way to validate 
+the quote, and a TCG Event Log (Linux only).
+
+Use --key to specify the type of attestation key. It can be gceAK for GCE attestation
+key or AK for a custom attestation key.By default it uses gceAK.
+--algo flag override the public key algorithm for attestation key. If not provided then
```

Comment:
ECC is more performant and our template uses an algorithm that is required to be implemented by the spec: https://trustedcomputinggroup.org/resource/tcg-algorithm-registry/.

Reply:
--algo flag is already defined in go-tpm-tools/cmd/flags.go and it takes rsa as a default value for --algo flag.